controllers/dspipeline_controller.go

```diff
@@ -304,7 +305,10 @@ func (r *DSPAReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 
 		err = r.ReconcileMLMD(dspa, params)
 		if err != nil {
+			r.setStatusAsNotReady(config.EnvoyReady, err, dspaStatus.SetEnvoyStatus)
 			return ctrl.Result{}, err
+		} else {
+			r.setStatus(ctx, "ds-pipeline-metadata-envoy-"+dspa.Name, config.EnvoyReady, dspa, dspaStatus.SetEnvoyStatus, log)
 		}
```

Comment:
no need for the `else` because you're returning above it when there's an error

although i see it's done like this everywhere else in here :(

```suggestion
		}
		r.setStatus(ctx, "ds-pipeline-metadata-envoy-"+dspa.Name, config.EnvoyReady, dspa, dspaStatus.SetEnvoyStatus, log)
```